lib/utils/web-auth.js

```diff
@@ -0,0 +1,22 @@
+const EventEmitter = require('events')
+const { webAuthCheckLogin } = require('npm-profile')
+
+function webAuth (opener, initialUrl, doneUrl, opts) {
+  const doneEmitter = new EventEmitter()
+
+  const openPromise = opener(initialUrl, doneEmitter)
+  const webAuthCheckPromise = webAuthCheckLogin(doneUrl, { ...opts, cache: false })
+    .then(authResult => {
+      // cancel open prompt if it's present
+      doneEmitter.emit('abort')
+
+      return authResult
+    })
+
+  return Promise.all([openPromise, webAuthCheckPromise]).then(
```

Comment:
the Promise.all seems useless here; i think you can `await openPromise; return await webAuthCheckPromise` instead?

Reply:
Good idea :+1: